engine/src/main/java/org/terasology/engine/config/SystemConfig.java

```diff
@@ -81,7 +83,7 @@ public class SystemConfig extends AutoConfig {
 
     public final Setting<Locale> locale = setting(
             type(Locale.class),
-            defaultValue(Locale.getDefault(Category.DISPLAY)),
+            defaultValue(getAdjustedLocale()),
```

Comment:
Would it make sense to only do this if the locale cannot be found, e.g. because it's a weird one?
This would allow to keep regional differences like `en_US` vs `en_GB` or `en_NZ` while fixing the macos issue.